python/pyarrow/tests/test_array.py

```diff
@@ -3546,3 +3548,59 @@ def test_run_end_encoded_from_buffers():
     with pytest.raises(ValueError):
         pa.RunEndEncodedArray.from_buffers(ree_type, length, buffers,
                                            1, offset, children)
+
+
+def PyCapsule_IsValid(capsule, name):
+    return ctypes.pythonapi.PyCapsule_IsValid(ctypes.py_object(capsule), name) == 1
+
+
+@pytest.mark.parametrize(
+    'value_type',
+    [pa.uint8(), pa.uint32(), pa.int16(), pa.float32()]
+)
+def test_dlpack(value_type):
+    if Version(np.__version__) < Version("1.22.0"):
+        pytest.skip("No dlpack support in numpy versions older than 1.22.0.")
+    arr = pa.array([1, 2, 3], type=value_type)
+
+    DLTensor = arr.__dlpack__()
+    assert PyCapsule_IsValid(DLTensor, b"dltensor") is True
+
+    expected = np.array([1, 2, 3])
+    result = np.from_dlpack(arr)
+    np.testing.assert_array_equal(result, expected)
```

Comment:
`assert_array_equal` doesn't require the dtypes to be the same? We should also assert that part